rotary_utils/repair.py

```diff
@@ -622,105 +494,147 @@ def run_end_repair(long_read_filepath: str, assembly_fasta_filepath: str, assemb
+                                             repair_paths.end_repaired_contigs_filepath, length_thresholds,
+                                             cli_tool_settings_dict,
+                                             repair_paths.verbose_logfile, threads)
+
+    os.makedirs(repair_paths.circlator_logs, exist_ok=True)
+    shutil.move(os.path.join(repair_paths.linking_outdir_base, 'log_summary'), repair_paths.circlator_logs)
 
     # Check for contigs that could not be circularized
     if len(failed_contig_names) != 0:
-        if os.path.isdir(os.path.join(linking_outdir_base, 'troubleshooting')):
-            shutil.move(os.path.join(linking_outdir_base, 'troubleshooting'),
-                        os.path.join(output_dir, 'troubleshooting'))
-
-        if keep_failed_contigs is False:
-            logger.error(f'{len(failed_contig_names)} contigs could not be circularized. A partial output file '
-                         f'including successfully circularized contigs (and no linear contigs) is available at '
-                         f'{end_repaired_contigs_filepath} for debugging. Exiting with error status. See temporary '
-                         f'files and verbose logs for more details.')
-            sys.exit(1)
-
-        elif keep_failed_contigs is True:
-            logger.warning(f'{len(failed_contig_names)} contigs could not be circularized. The original (non-repaired) '
-                           f'versions of these contigs will be included in the final output file')
-            logger.warning(f'Names of contigs that could not be circularized: {", ".join(failed_contig_names)}')
-
-            # Get the non-repaired circular contigs and append them to the repaired contigs file
-            with open(end_repaired_contigs_filepath, 'a') as append_handle:
-                for record in subset_sequences(assembly_fasta_filepath, failed_contig_names):
-                    SeqIO.write(record, append_handle, 'fasta')
-        else:
-            error = ValueError(f'keep_failed_contigs should be a boolean True or False; you provided '
-                               f'{keep_failed_contigs}')
-            logger.error(error)
-            raise error
+        repair_handle_failed_contigs(assembly_info, repair_paths, failed_contig_names, keep_failed_contigs, output_dir)
 
     # Get the linear contigs and append them to the repaired contigs file
     # TODO - consider just getting all non-circular contigs regardless of whether they are in the assembly info file.
     #   This edit might require modification of the subset_sequences function and refactoring of code in several places,
     #   but I imagine it is closer to the behaviour that the user would expect.
-    linear_contig_names = parse_assembly_info_file(assembly_info_filepath, assembly_info_type, return_type='linear')
+    linear_contig_names = parse_assembly_info_file(assembly_info.assembly_info_filepath,
```

Comment:
Could this be moved higher up in the function and passed to `repair_no_circular_input_contigs`?